native/common/jp_convert.cpp

```diff
@@ -0,0 +1,644 @@
+/*
+ * Copyright 2020 nelson85.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <jpype.h>
+
+static jvalue bool2bool(char* c)
+{
+	jvalue v;
+	v.z = (*(uint8_t*) c) == 0 ? false : true;
```

Comment:
Why do you need to cast to a pointer first, and then dereference it? Shouldn't you use c++ like casts so it can be compile time expressions?

Reply:
Not sure if there is a performance difference.  I can recode if you feel strongly.    This is really just about doing reinterpret cast over and over.  It doesn't matter at all for bool because I going to force true or false, but for others  I am forcing signed to unsigned conversions.

I was planning a big test bench that exercised each of the type conversions, but decided to request a review as we have a bunch of people stuck over the numpy extra problem, so  decided to rush this out and then search for edge cases once we had agreement on direction..

Reply:
I updated the pattern to more C++ like